### PR TITLE
feat(grouping): Mark mobile strategy as "latest" to allow upgrading

### PR DESCRIPTION
**Pull request by jjbayer**
We only allow upgrading the grouping configuration of a project to the strategy marked "latest".

"latest" should now be `mobile`, but _only_ for organizations which have the appropriate feature flag turned on, so we need a project-specific (or at least organization-specific) endpoint to load grouping configs.

https://getsentry.atlassian.net/browse/INGEST-177

### CODE REVIEW

File: static/app/views/settings/projectIssueGrouping/index.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ type State = {
   groupingConfigs: EventGroupingConfig[] | null;
 } & AsyncView['state'];
 
-class ProjectDebugSymbols extends AsyncView<Props, State> {
+class ProjectIssueGrouping extends AsyncView<Props, State> {
```
Comment (priscilawebdev) [inline static/app/views/settings/projectIssueGrouping/index.tsx:28]: thanks for this 🙏

**Review by priscilawebdev — APPROVED** [review]
tested locally and everything seems fine! Thanks for doing this @jjbayer 👏

**Review by untitaker — APPROVED** [review]

**Review by untitaker — COMMENTED** [review]
I now actually wonder if we want to mark this as latest or if we should have a `latest_beta` flag in addition that is then handled by the UI

Otherwise you cannot upgrade from legacy to newstyle if your organization is in EA, you need to jump straight to mobile.

File: src/sentry/api/endpoints/project_grouping_configs.py [hunk 2]
```diff
@@ -21,10 +22,10 @@ def get(self, request, project):
 
         # As long as newstyle:2019-10-29 is the global "latest", allow orgs to upgrade
         # to mobile:2021-02-12 if the appropriate feature flag is on:
-        if latest == "newstyle:2019-10-29" and features.has(
+        if latest == DEFAULT_GROUPING_CONFIG and features.has(
```
Comment (jjbayer) [inline src/sentry/api/endpoints/project_grouping_configs.py:25]: The reason I hard-coded the strategies here is that the behavior depends on a specific feature flag. As soon as this feature is enabled for everyone, "latest" would now point to `BETA_GROUPING_CONFIG` for everyone. We could of course make it dependent on something more generic like `is_early_adopter()`.

Comment (untitaker) [reply src/sentry/api/endpoints/project_grouping_configs.py:25]: ah ok, yeah I still think this is fine as-is for now

we still need to rename the grouping strategy away from mobile to something else so having a central variable is good

**Pull request merged**